### PR TITLE
Update adamw optimizer with vectorized apis

### PR DESCRIPTION
**Pull request by abuccts**
__Description__

Update adamw optimizer to compute scaling factor and d2h copy in batch for low precision gradients.

### CODE REVIEW

**Review by wkcn — APPROVED** [review]
LGTM. Thanks Yifan!

File: dockerfile/torch1.14-cuda11.8.dockerfile [hunk 1]
```diff
@@ -36,7 +36,8 @@ RUN apt-get update && \
 
 ARG NUM_MAKE_JOBS=
 ENV PATH="${PATH}" \
-    LD_LIBRARY_PATH="/usr/local/lib:${LD_LIBRARY_PATH}"
+    LD_LIBRARY_PATH="/usr/local/lib:${LD_LIBRARY_PATH}" \
+    PYTHONOPTIMIZE=1
```
Comment (tocean) [inline dockerfile/torch1.14-cuda11.8.dockerfile:40]: Where is this environment variable used?

Comment (abuccts) [reply dockerfile/torch1.14-cuda11.8.dockerfile:40]: it's for python

**Review by tocean — APPROVED** [review]
LGTM

**Pull request merged**